src/HttpClient/Plugin/ServerSideRequestForgeryProtection/Options.php

```diff
@@ -0,0 +1,300 @@
+    }
+
+    /**
+     * Enable sending of credenitals
+     * This is potentially a security risk.
+     *
+     * @return Options
+     */
+    public function enableSendCredentials()
+    {
+        $this->sendCredentials = true;
+
+        return $this;
+    }
+
+    /**
+     * Disable sending of credentials.
+     *
+     * @return Options
+     */
+    public function disableSendCredentials()
+    {
+        $this->sendCredentials = false;
+
+        return $this;
+    }
+
+    /**
+     * Get pin DNS option.
+     *
+     * @return bool
+     */
+    public function getPinDns()
+    {
+        return $this->pinDns;
+    }
+
+    /**
+     * Enable DNS pinning.
+     *
+     * @return Options
+     */
+    public function enablePinDns()
+    {
+        $this->pinDns = true;
+
+        return $this;
+    }
+
+    /**
+     * Disable DNS pinning.
+     *
+     * @return Options
+     */
+    public function disablePinDns()
+    {
+        $this->pinDns = false;
+
+        return $this;
+    }
+
+    /**
+     * Checks if a specific value is in a list.
+     *
+     * @param $list   string
+     * @param $type   string
+     * @param $value
+     *
+     * @return bool
+     * @throws InvalidOptionException
+     */
+    public function isInList($list, $type, $value)
+    {
+        if (!in_array($list, array('whitelist', 'blacklist'), true)) {
+            throw new InvalidOptionException('Provided list "'.$list.'" must be "whitelist" or "blacklist"');
+        }
+
+        if (!array_key_exists($type, $this->$list)) {
+            throw new InvalidOptionException('Provided type "'.$type.'" must be "ip", "port", "domain" or "scheme"');
+        }
+
+        if (empty($this->{$list}[$type])) {
+            return 'whitelist' === $list;
+        }
+
+        //For domains, a regex match is needed
+        if ('domain' === $type) {
+            foreach ($this->{$list}[$type] as $domain) {
+                if (preg_match('/^'.$domain.'$/i', $value)) {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        return in_array($value, $this->{$list}[$type]);
+    }
+
+    /**
+     * Returns a specific list.
+     *
+     * @param $list string
+     * @param $type string optional
+     *
+     * @return array
+     * @throws InvalidOptionException
+     */
+    public function getList($list, $type = null)
+    {
+        if (!in_array($list, array('whitelist', 'blacklist'), true)) {
+            throw new InvalidOptionException('Provided list "'.$list.'" must be "whitelist" or "blacklist"');
+        }
+
+        if ($type !== null) {
+            if (!array_key_exists($type, $this->$list)) {
+                throw new InvalidOptionException('Provided type "'.$type.'" must be "ip", "port", "domain" or "scheme"');
+            }
+
+            return $this->{$list}[$type];
+        }
+
+        return $this->{$list};
+    }
+
+    /**
+     * Sets a list to the passed in array.
```

Comment:
to the passed what?

Reply:
Copy paste from safecurl, I will fix it later